lib/cdo/pardot.rb

```diff
@@ -241,6 +225,30 @@ def self.sync_contacts_with_pardot(config)
     log "Contact #{config[:operation_name]} pass completed. #{num_operations} total operations."
   end
 
+  # Do special transformation of some fields from database to Pardot
+  # @param src [Hash] Hash of database fields we are reading from
+  # @param dst [Hash] Hash of Pardot fields we are writing to
+  def self.apply_special_fields(src, dest)
+    # special case: if contact has opted out, set the two different Pardot
+    # flavors of opted out to true. Also, only ever set this to true, otherwise
+    # set no value; never set it back to false. Pardot is the authority on
+    # opt-out data, so never reset any opt-out setting it has stored.
+    if src[:opted_out]
+      dest[:opted_out] = true
+      dest[:is_do_not_email] = true
+    end
+
+    # If this contact has a dashboard user ID (which means it is a teacher
+    # account), mark that in a Pardot field so we can segment on that.
+    if src[:dashboard_user_id].present?
+      dest[:db_Has_Teacher_Account] = "true"
+    end
+
+    # set a custom field to mark in Pardot that this contact was imported from
+    # Code Studio
+    dest[:db_Imported] = "true"
```

Comment:
Though I probably asked already and forgot, why are these `"true"` whereas the opt out fields are `true`?